### PR TITLE
fix: integrated with pl's device.use_grouping to fix a bug with qaoa …

### PR DESCRIPTION
**Pull request by jacofeld**
…ing multiple observables

*Issue #, if available:*

*Description of changes:*
Integrated with PennyLane's `device.use_grouping` property to disable grouping when Braket computes the gradient using AdjointGradient. Grouping needs to be disabled in that case because the output of it is multiple observables, and Braket doesn't support computing gradients with multiple observables. See [this pr](https://github.com/PennyLaneAI/pennylane/pull/3456) that added the `use_grouping` property to PennyLane.

*Testing done:*
Added unit tests for the new method.
Added an integration test with the QAOA steps from that qchem module that were failing originally.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by speller26** [comment]
Does grouping still happen even if `optimize` isn't set to `True` in the qnode? I thought grouping had to be requested.

**Review by kshyatt-aws — DISMISSED** [review]

File: setup.py [hunk 1]
```diff
@@ -61,6 +61,7 @@
             "tox",
             "tensorflow>=2.6.0",
             "torch>=1.11",
+            "openfermionpyscf",
```
Comment (jacofeld) [inline setup.py:64]:
```suggestion

```

File: test/integ_tests/conftest.py [hunk 2]
```diff
@@ -86,6 +86,11 @@
 # pytest fixtures
 
 
+@pytest.fixture
+def qchem_data_dir():
+    return os.path.join(os.path.dirname(__file__), "qchem")
+
+
```
Comment (jacofeld) [inline test/integ_tests/conftest.py:93]:
```suggestion

```

**Comment by jacofeld** [comment]
> Does grouping still happen even if `optimize` isn't set to `True` in the qnode? I thought grouping had to be requested.

Yes, PennyLane's QAOA module [enables grouping](https://github.com/PennyLaneAI/pennylane/blob/0d8a73b7a072eab2ab03615a26d773b4b9fb6063/pennylane/qaoa/cost.py#L391), and we can't disable it without without `use_grouping`. This caused some errors in our qaoa notebook as it was originally written

**Comment by josh146** [comment]
@speller26 this was a performance 'hotfix' we made documented here: https://github.com/PennyLaneAI/pennylane/pull/1718. The better longer term fix is to allow devices to have more of a say of when/where grouping is used

**Review by speller26 — APPROVED** [review]

**Pull request merged**